src/idl_parser.cpp

```diff
@@ -1120,6 +1127,39 @@ CheckedError Parser::ParseVector(const Type &type, uoffset_t *ovalue) {
   return NoError();
 }
 
+CheckedError Parser::ParseNestedFlatbuffer(Value &val, FieldDef *field,
+                                          size_t fieldn,
+                                          const StructDef *parent_struct_def) {
+  if (token_ == '[') {// backwards compat for 'legacy' ubyte buffers
+    ECHECK(ParseAnyValue(val, field, fieldn, parent_struct_def));
+  } else {
+    auto cursor_at_value_begin = cursor_;
+    ECHECK(SkipAnyJsonValue());
+    std::string substring(cursor_at_value_begin -1 , cursor_ -1);
+
+    //create and initialize new parser
+    Parser nested_parser;
+    assert(field->nested_flatbuffer);
+    nested_parser.root_struct_def_ = field->nested_flatbuffer;
+    nested_parser.enums_ = enums_;
+    nested_parser.opts = opts;
+    nested_parser.uses_flexbuffers_ = uses_flexbuffers_;
+    //done initializing
```

Comment:
This kind of comment doesn't really say anything.
In general, comments should be: `// Like this.` not `//like this`, and only when they say something new. No need for empty lines before/after comments.